File: runtime/CSharp/Antlr4.Runtime/Atn/RangeTransition.cs
// Copyright (c) Terence Parr, Sam Harwell. All Rights Reserved.
// Licensed under the BSD License. See LICENSE.txt in the project root for license information.

using Antlr4.Runtime.Misc;
using Antlr4.Runtime.Sharpen;

namespace Antlr4.Runtime.Atn
{
    public sealed class RangeTransition : Transition
    {
        public readonly int from;

        public readonly int to;

        public RangeTransition(ATNState target, int from, int to)
            : base(target)
        {
            this.from = from;
            this.to = to;
        }

        public override Antlr4.Runtime.Atn.TransitionType TransitionType
        {
            get
            {
                return Antlr4.Runtime.Atn.TransitionType.Range;
            }
        }

        public override IntervalSet Label
        {
            get
            {
                return IntervalSet.Of(from, to);
            }
        }

        public override bool Matches(int symbol, int minVocabSymbol, int maxVocabSymbol)
        {
            return symbol >= from && symbol <= to;
        }

        [return: NotNull]
        public override string ToString()
        {
            return "'" + (char)from + "'..'" + (char)to + "'";
        }
    }
}

File: reference/antlr4/runtime/Java/src/org/antlr/v4/runtime/atn/RangeTransition.java
/*
 * Copyright (c) 2012 The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD-3-Clause license that
 * can be found in the LICENSE.txt file in the project root.
 */

package org.antlr.v4.runtime.atn;

import org.antlr.v4.runtime.misc.IntervalSet;
import org.antlr.v4.runtime.misc.NotNull;

public final class RangeTransition extends Transition {
	public final int from;
	public final int to;

	public RangeTransition(@NotNull ATNState target, int from, int to) {
		super(target);
		this.from = from;
		this.to = to;
	}

	@Override
	public TransitionType getSerializationType() {
		return TransitionType.RANGE;
	}

	@Override
	@NotNull
	public IntervalSet label() { return IntervalSet.of(from, to); }

	@Override
	public boolean matches(int symbol, int minVocabSymbol, int maxVocabSymbol) {
		return symbol >= from && symbol <= to;
	}

	@Override
	@NotNull
	public String toString() {
		return "'"+(char)from+"'..'"+(char)to+"'";
	}
}
